src/iologindata.cpp

```diff
@@ -52,12 +52,40 @@ bool IOLoginData::saveAccount(const Account& acc)
 	return Database::getInstance()->executeQuery(query.str());
 }
 
+std::string decodeSecret(const std::string& secret) {
+	// simple base32 decoding
+	std::string key;
+	key.reserve(10);
+
+	uint32_t buffer = 0, left = 0;
+	for (const auto& ch : secret) {
+		buffer <<= 5;
+		if (ch >= 'A' && ch <= 'Z') {
+			buffer |= (ch & 0x1F) - 1;
+		} else if (ch >= '2' && ch <= '7') {
+			buffer |= ch - 24;
+		} else {
+			// if a key is broken, return empty and the comparison
+			// will always be false since the token must not be empty
+			return {};
```

Comment:
It's C++11 syntax, instead of `return ns::thing(arg1, arg2, arg3...)`  you can just `return {arg1, arg2, arg3...}` provided that the return type is `ns::thing`.